# Generated by Django 3.2.18 on 2023-04-21 19:56

from django.db import migrations, models
import django.db.models.deletion


def populate_refundable_amount(apps, schema_editor):
    Payment = apps.get_model("payments", "Payment")

    payments = list()
    for payment in Payment.objects.all():
        payment.refundable_amount = payment.amount
        payments.append(payment)

    Payment.objects.bulk_update(payments, ['refundable_amount'])


class Migration(migrations.Migration):

    dependencies = [
        ('invoices', '0007_invoicelineitem'),
        ('payments', '0003_payment_check_number'),
    ]

    operations = [
        migrations.AddField(
            model_name='payment',
            name='refundable_amount',
            field=models.DecimalField(decimal_places=2, default=0, max_digits=15),
        ),
        migrations.CreateModel(
            name='Refund',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('response_status', models.CharField(choices=[('pending', 'Pending'), ('succeeded', 'Succeeded'), ('failed', 'Failed')], default='pending', max_length=50)),
                ('response_reason_text', models.TextField(default='')),
                ('error_message', models.TextField(blank=True, null=True)),
                ('trans_id', models.CharField(max_length=100)),
                ('refund_id', models.CharField(blank=True, max_length=100, null=True)),
                ('amount', models.DecimalField(decimal_places=2, max_digits=15)),
                ('transaction_dt', models.DateTimeField(auto_now_add=True)),
                ('invoice', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='invoices.invoice')),
                ('payment', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='payments.payment')),
            ],
        ),
        migrations.RunPython(populate_refundable_amount, migrations.RunPython.noop)
    ]
